Formula/libff.rb

```diff
@@ -0,0 +1,50 @@
+class Libff < Formula
+  desc "C++ library for Finite Fields and Elliptic Curves"
+  homepage "https://github.com/scipr-lab/libff"
+  # pull from git tag to get submodules
+  url "https://github.com/scipr-lab/libff.git",
+      tag:      "v0.2.1",
+      revision: "5835b8c59d4029249645cf551f417608c48f2770"
+  license "MIT"
+
+  depends_on "cmake" => :build
+  depends_on "openssl@1.1" => :build
+
+  depends_on "gmp"
+
+  def install
+    # bn128 is somewhat faster, but requires an x86_64 CPU
+    curve = Hardware::CPU.intel? ? "BN128" : "ALT_BN128"
+
+    # build libff dynamically. The project only builds statically by default
+    inreplace "libff/CMakeLists.txt", "STATIC", "SHARED"
+    inreplace "depends/CMakeLists.txt", "endif()", <<~'EOS'
+        set_property(TARGET zm PROPERTY POSITION_INDEPENDENT_CODE ON)
+      endif()
+    EOS
```

Comment:
Can't we just set `CMAKE_POSITION_INDEPENDENT_CODE=ON`?

Reply:
It's certainly a more elegant way to go about it 👍